pkg/bootstrap/docker/up.go

```diff
@@ -0,0 +1,647 @@
+package docker
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/blang/semver"
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	kclientcmd "k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/openshift/origin/pkg/bootstrap/docker/dockerhelper"
+	"github.com/openshift/origin/pkg/bootstrap/docker/dockermachine"
+	"github.com/openshift/origin/pkg/bootstrap/docker/errors"
+	"github.com/openshift/origin/pkg/bootstrap/docker/host"
+	"github.com/openshift/origin/pkg/bootstrap/docker/openshift"
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	osclientcmd "github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	dockerutil "github.com/openshift/origin/pkg/cmd/util/docker"
+)
+
+const (
+	// RecommendedName is the recommended command name
+	CmdUpRecommendedName = "up"
+
+	openShiftContainer = "origin"
+	openShiftNamespace = "openshift"
+
+	initialUser     = "developer"
+	initialPassword = "developer"
+
+	initialProjectName    = "myproject"
+	initialProjectDisplay = "My Project"
+	initialProjectDesc    = "Initial developer project"
+
+	cmdUpLong = `
+Starts an OpenShift cluster using Docker containers, provisioning a registry, router,
+initial templates, and a default project.
+
+This command will attempt to use an existing connection to a Docker daemon. Before running
+the command, ensure that you can execure docker commands successfully (ie. 'docker ps').
+
+Optionally, the command can create a new Docker machine for OpenShift using the VirtualBox
+driver when the --create-machine argument is specified. The machine will be named 'openshift'
+by default. To name the machine differently, use the --docker-machine=NAME argument. If the
+--docker-machine=NAME argument is specified, but --create-machine is not, the command will attempt
+to find an existing docker machine with that name and start it if it's not running.
+
+By default, the OpenShift cluster will be setup to use a routing suffix that ends in xip.io.
+This is to allow dynamic host names to be created for routes. An alternate routing suffix
+can be specified using the --routing-suffix flag.
+
+A public hostname can also be specified for the server with the --public-hostname flag.
+`
+	cmdUpExample = `
+  # Start OpenShift on a new docker machine named 'openshift'
+  %[1]s --create-machine
+
+  # Start OpenShift using a specific public host name
+  %[1]s --public-hostname=my.address.example.com
+  
+  # Start OpenShift and preserve data and config between restarts
+  %[1]s --host-data-dir=/mydata --use-existing-config
+`
+)
+
+var (
+	imageStreamLocations = map[string]string{
+		"origin centos7 image streams": "examples/image-streams/image-streams-centos7.json",
+	}
+	templateLocations = map[string]string{
```

Comment:
Will this fail if I don't have these directories?

What's left to consume these?